1.process_data/nbconverted/process_and_normalize_jump_data.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Preprocess using pyctyominer and merge broad samples
+
+# ## Imports
+
+# In[ ]:
+
+
+import time
+from pathlib import Path
+import numpy as np
+import pandas as pd
+from pycytominer import normalize
+from pycytominer.cyto_utils.cells import SingleCells
+
+
+# ## Find the root of the git directory
+# This allows file paths to be referenced in a system agnostic way
+
+# In[ ]:
+
+
+# Get the current working directory
+cwd = Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Define Paths
+
+# In[ ]:
+
+
+# Input paths
+big_drive_path = f"{root_dir}/big_drive"
+sqlite_data_path = f"{big_drive_path}/data"
+ref_path = f"{root_dir}/reference_plate_data"
+barcode_platemap = f"{ref_path}/barcode_platemap.csv"
+
+# Output paths
+output_cell_count_path = Path(f"{big_drive_path}/sc_counts")
+normalized_path = Path(f"{big_drive_path}/normalized_sc_data")
+
+
+# ## Create directories if non-existent
+
+# In[ ]:
+
+
+output_cell_count_path.mkdir(parents=True, exist_ok=True)
+normalized_path.mkdir(parents=True, exist_ok=True)
+
+
+# In[ ]:
+
+
+# Create dataframe from barcode platemap
+barcode_df = pd.read_csv(barcode_platemap)
+
+
+# # Process cell data
+
+# ## Map reference data
+
+# In[ ]:
+
+
+# Merge on the broad_sample column
+merge_col = "Metadata_broad_sample"
+
+compdf = pd.read_csv(f"{ref_path}/JUMP-Target-1_compound_metadata_targets.tsv", sep="\t")
+
+# Set empty broad samples to DMSO if the pert iname is DMSO for the compounds dataframe
+compdf.loc[compdf["pert_iname"] == "DMSO", "broad_sample"] = "DMSO"
+
+# Map platemap names found in the barcode file to metadata dataframes
+barcode_map = {"JUMP-Target-1_orf_platemap": pd.read_csv(f"{ref_path}/JUMP-Target-1_orf_metadata.tsv", sep="\t"),
+               "JUMP-Target-1_crispr_platemap": pd.read_csv(f"{ref_path}/JUMP-Target-1_crispr_metadata.tsv", sep="\t"),
+                "JUMP-Target-1_compound_platemap": compdf}
+
+# Map platemap names found in the barcode file to platemap dataframes
+platemeta2df = {platemap_name: pd.read_csv(f"{ref_path}/{platemap_name}.txt", sep="\t") for platemap_name, _ in barcode_map.items()}
+
+# Map the wells corresponding to the empty broad samples in the plate metadata files
+platemeta2cols = {name: df.loc[df["broad_sample"].isnull()]["well_position"].tolist() for name, df in platemeta2df.items() if name != "JUMP-Target-1_compound_platemap"}
+
+
+# ## Rename columns and fill control values
+
+# In[ ]:
+
+
+def rename_columns(df):
```

Comment:
```suggestion
def add_metadata_prefix_to_column_names(df):
```

I recommend using very explicit function names :)

Reply:
Changed!